client/src/components/LoginModal.vue

```diff
@@ -0,0 +1,145 @@
+
+<script lang="ts">
+import { defineComponent, ref } from "vue"
+import { defineEmits } from "vue"
+
+export default defineComponent({
+    name: "LoginModal",
+    setup() {
```

Comment:
You can simplify this by using ``script setup``. See https://vuejs.org/api/sfc-script-setup.html#script-setup.